streams/streams-scala/src/main/scala/org/apache/kafka/streams/scala/Serdes.scala

```diff
@@ -30,12 +32,15 @@ object Serdes {
   implicit def JavaLong: Serde[java.lang.Long] = JSerdes.Long()
   implicit def ByteArray: Serde[Array[Byte]] = JSerdes.ByteArray()
   implicit def Bytes: Serde[org.apache.kafka.common.utils.Bytes] = JSerdes.Bytes()
+  implicit def byteBufferSerde: Serde[ByteBuffer] = JSerdes.ByteBuffer()
+  implicit def shortSerde: Serde[Short] = JSerdes.Short().asInstanceOf[Serde[Short]]
   implicit def Float: Serde[Float] = JSerdes.Float().asInstanceOf[Serde[Float]]
   implicit def JavaFloat: Serde[java.lang.Float] = JSerdes.Float()
   implicit def Double: Serde[Double] = JSerdes.Double().asInstanceOf[Serde[Double]]
   implicit def JavaDouble: Serde[java.lang.Double] = JSerdes.Double()
   implicit def Integer: Serde[Int] = JSerdes.Integer().asInstanceOf[Serde[Int]]
   implicit def JavaInteger: Serde[java.lang.Integer] = JSerdes.Integer()
+  implicit def uuidSerde: Serde[UUID] = JSerdes.UUID()
```

Comment:
Given the naming of the other implicits, should it be
```
byteBufferSerde -> ByteBuffer
shortSerde -> Short
uuidSerde -> UUID
```

Reply:
Agreed, although it's objectively terrible for us to name the serde supplier exactly the same as the type it serializes, we might as well keep the convention going.

Reply:
Would it be ok to rename all? Or would this break something?

Reply:
If people import specific implicits, it'd break them. Probably better leave it alone, unfortunately.

Reply:
It's definitely a problem, but I think it should be discussed separately. Because changing names of implicits will break source and binary compatibility.

Reply:
```suggestion
  implicit def UUID: Serde[UUID] = JSerdes.UUID()
```

Agreed. Thanks for the response, Yuriy!

Just to be clear, I've made github "code suggestions" for the things that Matthias was asking for. Sorry if my side remark confused the issue. We should stick with the convention of the other implicits, even though it's a bummer.

Reply:
I'm **strongly** against these fixes. It's not a "convention", it's just a mistake of the original author of this code. And this mistake is not illusory, I faced with it on the first day of using kafka streams api. Wildcard import of `org.apache.kafka.streams.scala.Serdes._` breaks the code because of the name clash. For example:
```scala
import org.apache.kafka.streams.scala.Serdes._

val x = String.valueOf(5) // Error:(7, 18) value valueOf is not a member of org.apache.kafka.common.serialization.Serde[String]
println(x)
```
I'm absolutely sure that it's more important to not introduce more broken things in the new code than follow mistakenly introduced convention. Yes, it will look a bit strange, but it's a way less important. 

I planned to propose a fix for this naming issue in the separate pull request. Let's leave the current pull request without changes and discuss implicit names issue separately?

Reply:
Valid point. However, for consistency reasons, I would like to not introduce two name patterns in parallel.

To rename the existing implicits, we would need to deprecate the existing ones and add new ones with new name on the side. Having the old naming pattern deprecated and adding a new naming pattern make a clean cut.

This change requires a KIP though (https://cwiki.apache.org/confluence/display/KAFKA/Kafka+Improvement+Proposals) -- It should be non-controversial and thus if we start the KIP right away, we can meet the upcoming KIP deadline for 2.6. release.

Would you be interested to do a KIP?

Reply:
Well, I can try.

About this pull request — if you insist to use old naming, I prefer to postpone this pull request until naming will be fixed. I don't wont to introduce more problems for users.

Reply:
I am fine with postponing this PR. \cc @vvcephei

Reply:
I created the KIP: https://cwiki.apache.org/confluence/display/KAFKA/KIP-616+-+Rename+implicit+Serdes+instances+in+kafka-streams-scala

Reply:
Agreed, It seems like a good idea actually just to close this PR in favor of the KIP (which I think is great), and a new PR.

Reply:
We can also reuse this PR. Just update the title and "ready to go". I don't care. Whatever @LMnet prefers

Reply:
I decided to create a [new pull request](https://github.com/apache/kafka/pull/8955).